src/invoice2data/extract/invoice_template.py

```diff
@@ -151,12 +154,23 @@ def coerce_type(self, value, target_type):
             return self.parse_date(value)
         assert False, "Unknown type"
 
-    def extract(self, optimized_str):
+    def extract(self, optimized_str: str, invoice_file: str, input_module: str) -> Optional[dict]:
         """
         Given a template file and a string, extract matching data fields.
-        """
 
-        logger.debug("START optimized_str ========================\n" + optimized_str)
```

Comment:
Could you leave the debugger output i this form? It prevents the first line from being misaligned.